src/Datadog.Trace/Util/ArrayHelper.cs

```diff
@@ -7,8 +9,50 @@ public static T[] Empty<T>()
 #if NET45
             return EmptyArray<T>.Value;
 #else
-            return System.Array.Empty<T>();
+            return Array.Empty<T>();
+#endif
+        }
+
+        public static T[] Copy<T>(T[] array)
+        {
+            int length = array.Length;
+
+            return Copy(array, length, length);
+        }
+
+        public static T[] Copy<T>(T[] array, int newSize, int length)
+        {
+            var target = new T[newSize];
+            Copy(array, target, length);
+
+            return target;
+        }
+
+        public static void Copy<T>(T[] source, T[] target, int length)
+        {
+            Copy(source, target, 0, 0, length);
+        }
+
+        public static void Copy<T>(T[] source, T[] target, int sourcOffset, int targetOffset, int length)
```

Comment:
nit: typo in `sourcOffset` variable name

Reply:
Small API feedback, I recommend changing the order of the input parameters to `T[] source, int sourceOffset, T[] target, int targetOffset, int length` to match the `Array.Copy` and `Buffer.BlockCopy` APIs

Reply:
If this was already the shape of the Jaeger ArrayHelper class then no worries, no need to act on this feedback 😄

Reply:
Agree, changed.